plonk/src/proof_system/prover.rs

```diff
@@ -783,11 +783,11 @@ impl<E: Pairing> Prover<E> {
         let n = pk.domain_size();
         let m = self.quot_domain.size();
         let domain_size_ratio = m / n;
-        let n_field = E::ScalarField::from(n as u64);
-        let lagrange_n_coeff =
-            self.domain.group_gen_inv / (n_field * (eval_point - self.domain.group_gen_inv));
-        let lagrange_1_coeff =
-            E::ScalarField::one() / (n_field * (eval_point - E::ScalarField::one()));
+        let vanish_eval = self.domain.evaluate_vanishing_polynomial(eval_point);
```

Comment:
Would this be more expensive than the previous code?

Reply:
i think so, because we `* vanish_eval` inside `lagrange_1/n_coeff()`, then divide it outside.. 🤔

Reply:
here's the cheaper alternative: [4fdde7c](https://github.com/EspressoSystems/jellyfish/pull/611/commits/4fdde7c42e366f27f828b1be4c124a579bc0380f) 

lmk which do you prefer

Reply:
after discussion, we will revert this change and go for the more readable option

Reply:
changed back in [7f27a78](https://github.com/EspressoSystems/jellyfish/pull/611/commits/7f27a78c19f146bfd35167ec2719732f274fdb85)